tests/bluetooth/audio/cap_commander/src/test_cancel.c

```diff
@@ -0,0 +1,183 @@
+		fixture->start_member_params[i].broadcast_id = 0;
+		memcpy(fixture->start_member_params[i].subgroups, &fixture->subgroups[0],
+		       sizeof(struct bt_bap_bass_subgroup) * CONFIG_BT_BAP_BASS_MAX_SUBGROUPS);
+		fixture->start_member_params[i].num_subgroups = CONFIG_BT_BAP_BASS_MAX_SUBGROUPS;
+	}
+
+	for (size_t i = 0; i < ARRAY_SIZE(fixture->conns); i++) {
+		err = bt_cap_commander_discover(&fixture->conns[i]);
+		zassert_equal(0, err, "Unexpected return value %d", err);
+	}
+}
+
+static void
+cap_commander_test_cancel_fixture_init(struct cap_commander_test_cancel_fixture *fixture)
+{
+	for (size_t i = 0; i < ARRAY_SIZE(fixture->conns); i++) {
+		test_conn_init(&fixture->conns[i]);
+		fixture->conns[i].index = i;
+	}
+
+	test_start_param_init(fixture);
+}
+
+static void *cap_commander_test_cancel_setup(void)
+{
+	struct cap_commander_test_cancel_fixture *fixture;
+
+	fixture = malloc(sizeof(*fixture));
+	zassert_not_null(fixture);
+
+	return fixture;
+}
+
+static void cap_commander_test_cancel_before(void *f)
+{
+	int err;
+	struct cap_commander_test_cancel_fixture *fixture = f;
+
+	memset(f, 0, sizeof(struct cap_commander_test_cancel_fixture));
+	cap_commander_test_cancel_fixture_init(fixture);
+
+	for (size_t i = 0; i < ARRAY_SIZE(fixture->conns); i++) {
+		err = bt_cap_commander_discover(&fixture->conns[i]);
+		zassert_equal(0, err, "Unexpected return value %d", err);
+	}
+}
+
+static void cap_commander_test_cancel_after(void *f)
+{
+	struct cap_commander_test_cancel_fixture *fixture = f;
+
+	bt_cap_commander_unregister_cb(&mock_cap_commander_cb);
+
+	for (size_t i = 0; i < ARRAY_SIZE(fixture->conns); i++) {
+		mock_bt_conn_disconnected(&fixture->conns[i], BT_HCI_ERR_REMOTE_USER_TERM_CONN);
+	}
+}
+
+static void cap_commander_test_cancel_teardown(void *f)
+{
+	free(f);
+}
+
+static void test_cancel(void)
+{
+	int err;
+
+	err = bt_cap_commander_cancel();
+	zassert_equal(0, err, "Unexpected return value %d", err);
+
+	zexpect_call_count("bt_cap_commander_cb.broadcast_reception_start", 1,
+			   mock_cap_commander_broadcast_reception_start_cb_fake.call_count);
+	zassert_equal(-ECANCELED,
+		      mock_cap_commander_broadcast_reception_start_cb_fake.arg1_history[0]);
+}
+
+ZTEST_SUITE(cap_commander_test_cancel, NULL, cap_commander_test_cancel_setup,
+	    cap_commander_test_cancel_before, cap_commander_test_cancel_after,
+	    cap_commander_test_cancel_teardown);
+
+ZTEST_F(cap_commander_test_cancel, test_commander_cancel)
+{
+	int err;
+
+	err = bt_cap_commander_register_cb(&mock_cap_commander_cb);
+	zassert_equal(0, err, "Unexpected return value %d", err);
+
+	/* Do not run the add_src callback, so that the broadcast reception start procedure does not
+	 * run until completion
+	 */
+	set_skip_add_src(1);
+
+	/* initiate a CAP procedure; for this test we use broadcast reception start*/
+	err = bt_cap_commander_broadcast_reception_start(&fixture->start_param);
+	zassert_equal(0, err, "Could not start CAP procedure: %d", err);
```

Comment:
Is there value in testing the cancel for the other CAP Commander procedures too, and not just the broadcast reception start?

Reply:
I don't think so, there is nothing in the cancel command that relates to a specific CAP command

Reply:
I think you are right. All CAP Commander procedures depend on write responses IIRC